File: JobsAPI_Ruby/job.rb
module Jobs
  class Job
    def initialize(deviceId, jobId, jobName, jobProgress, jobSubmitTime,jobLastEventTime, jobType,parentJobId,jobCompleteTime, jobWorkTimeEstimate, location, marker)
      @deviceId = deviceId
      @jobId = jobId
      @jobName = jobName
      @jobProgress = jobProgress
      @jobSubmitTime = jobSubmitTime
      @jobType = jobType
      @jobWorkTimeEstimate = jobWorkTimeEstimate
      @location = location
      @marker = marker
      @jobLastEventTime = jobLastEventTime
      @parentJobId = parentJobId
      @jobCompleteTime = jobCompleteTime
    end

    attr_accessor :deviceId
    :jobId
    :jobName
    :jobProgress
    :jobSubmitTime
    :jobType
    :jobWorkTimeEstimate
    :jobLastEventTime
    :parentJobId
    :location
    :marker
    :jobCompleteTime
  end
  class JobComparator
    def Equals (job1, job2)
      return job1.jobId == job2.jobId && job1.deviceId == job2.deviceId
    end
  end
end

File: JobsAPI_C#/Jobs/Jobs API/Job.cs
using System;
using System.Collections.Generic;

namespace Jobs
{
    class Job
    {
        public string deviceId { get; set; }
        public string jobId { get; set; }
        public string jobName { get; set; }
        public string jobProgress { get; set; }
        public string jobSubmitTime { get; set; }
        public string jobType { get; set; }
        public int jobWorkTimeEstimate { get; set; }
        public string location { get; set; }
        public long marker { get; set; }
        public string ticketTemplate { get; set; }
    }

    class JobComparator : IEqualityComparer<Job>
    {
        public bool Equals(Job x, Job y)
        {
            if (Object.ReferenceEquals(x, y)) return true;
            if (Object.ReferenceEquals(x, null) || Object.ReferenceEquals(y, null))
                return false;

            return x.jobId == y.jobId && x.deviceId == y.deviceId;
        }

        public int GetHashCode(Job job)
        {
            //Check whether the object is null
            if (Object.ReferenceEquals(job, null)) return 0;

            int hashJobId = job.jobId.GetHashCode();
            int hashDeviceId = job.deviceId.GetHashCode();

            //Calculate the hash code for the product.
            return hashJobId ^ hashDeviceId;
        }
    }
}
